utils/pattern/src/implementations.rs

```diff
@@ -0,0 +1,15 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use alloc::borrow::Cow;
+use zerovec::{maps::ZeroMapKV, ZeroVec};
+
+use crate::{Pattern, SinglePlaceholder};
+
+impl<'data> ZeroMapKV<'data> for Pattern<SinglePlaceholder, Cow<'data, str>> {
```

Comment:
@sffc and @Manishearth 

I would like if you can give me your help in this implementation.

Reply:
is it should be something like this ?
```
impl<'data> ZeroMapKV<'data> for Pattern<SinglePlaceholder, Cow<'data, str>> {
    type Container = ZeroVec<Pattern<SinglePlaceholder, Cow<'data, str>>>;
    type Slice = ZeroVec<Pattern<SinglePlaceholder, Cow<'data, str>>>;
    type GetType = Cow<'data, str>;
    type OwnedType = Cow<'data, str>;
}

```

Reply:
It might be more like

```rust
impl<'data> ZeroMapKV<'data> for Pattern<SinglePlaceholder, str> {
    type Container = VarZeroVec<'data, Pattern<SinglePlaceholder, str>>;
    type Slice = VarZeroSlice<Pattern<SinglePlaceholder, str>>;
    type GetType = Pattern<SinglePlaceholder, str>;
    type OwnedType = Pattern<SinglePlaceholder, String>;
}
```

Reply:
There are still some errors, I will try to fix them in the morning.